config/dependency.go

```diff
@@ -273,10 +282,26 @@ func getTerragruntOutput(dependencyConfig Dependency, terragruntOptions *options
 		return nil, true, errors.WithStackTrace(DependencyConfigNotFound{Path: targetConfig})
 	}
 
-	jsonBytes, err := runTerragruntOutputJson(terragruntOptions, targetConfig)
-	if err != nil {
-		return nil, true, err
+	// Acquire synchronization lock to ensure only one instance of output is called per config.
+	actualLock, _ := outputLocks.LoadOrStore(targetConfig, &sync.Mutex{})
```

Comment:
Perhaps this approach will allow you to not have to cast it twice below?

```suggestion
	actualLock, _ := outputLocks.LoadOrStore(targetConfig, &sync.Mutex{}).(*sync.Mutex)
```

Reply:
Also, sanity check: I assume `targetConfig` is a *canonical* file system path?

Reply:
You can't do that because `LoadOrStore` returns two values.

> I assume targetConfig is a canonical file system path?

You're right. It wasn't before. Fixed: 8a97525